templates/flavors/private/patches/ilb-iptables-cmd.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+kind: KubeadmControlPlane
+metadata:
+  name: "${CLUSTER_NAME}-control-plane"
+spec:
+  kubeadmConfigSpec:
+    preKubeadmCommands:
+      # this is a workaround for the hairpin routing problem with internal load balancers
+      # during kubeadm init where 10.128.0.100 is the internal LB IP.
+      - if [ -f /tmp/kubeadm.yaml ]; then iptables -t nat -A OUTPUT -p all -d 10.128.0.100 -j DNAT --to-destination 127.0.0.1; fi
```

Comment:
`if [ -f /tmp/kubeadm.yaml ]` should be replaced by https://github.com/kubernetes-sigs/cluster-api/issues/3828 in the future